core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -1094,6 +1119,8 @@ private[log] class CleanedTransactionMetadata {
 
   // Output cleaned index to write retained aborted transactions
   var cleanedIndex: Option[TransactionIndex] = None
+  
+  var toAppend: List[AbortedTxn] = List.empty
```

Comment:
Can you help me understand why we needed to change the logic for appending to the transaction index?

Reply:
I believe the issue is that this code expects the log to be created already but it is not. I think my intention was to delay this operation until the log is created. I'll double check this is actually the case.